test/unit_test.c

```diff
@@ -363,32 +380,49 @@ static void test_mqtt(void) {
   }
 
   // Connect with empty client ID
-  c = mg_mqtt_connect(&mgr, url, NULL, mqtt_cb, buf);
-  for (i = 0; i < 200 && buf[0] == 0; i++) mg_mgr_poll(&mgr, 10);
+  c = mg_mqtt_connect(&mgr, url, NULL, mqtt_cb, &test_data);
+  for (i = 0; i < 300 && buf[0] == 0; i++) mg_mgr_poll(&mgr, 10);
   if (buf[0] != 'X') MG_INFO(("[%s]", buf));
   ASSERT(buf[0] == 'X');
+  ASSERT(test_data.subscribed == 0);
   mg_mqtt_sub(c, topic, 1);
+  for (i = 0; i < 500 && test_data.subscribed == 0; i++) mg_mgr_poll(&mgr, 10);
+  ASSERT(test_data.subscribed == 1);
+  ASSERT(test_data.published == 0);
   mg_mqtt_pub(c, topic, data, 1, false);
-  for (i = 0; i < 300 && buf[1] == 0; i++) mg_mgr_poll(&mgr, 10);
+  for (i = 0; i < 500 && test_data.published == 0; i++) mg_mgr_poll(&mgr, 10);
+  ASSERT(test_data.published == 1);
+  for (i = 0; i < 500 && buf[1] == 0; i++) mg_mgr_poll(&mgr, 10);
   if (strcmp(buf, "Xx/f12/hi") != 0) MG_INFO(("[%s]", buf));
   ASSERT(strcmp(buf, "Xx/f12/hi") == 0);
 
   // Set params
+  test_data.subscribed = 0;
+  test_data.published = 0;
   memset(buf, 0, sizeof(buf));
   memset(&opts, 0, sizeof(opts));
   opts.clean = true;
   opts.will_qos = 1;
   opts.will_retain = true;
   opts.keepalive = 20;
-  opts.client_id = mg_str("mg_client");
```

Comment:
Can't hard code the client_id here since tests run against the same public mqtt service in parallel causing breakage when the client_id's conflict.